src/seedu/addressbook/Main.java

```diff
@@ -107,13 +107,23 @@ private CommandResult executeCommand(Command command)  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
+            checkStorageFileExist();
```

Comment:
You shouldn't checked the existence of the file here, this doesn't satisfy SLAP. Better to check inside save method. Also consider creating the file and show the problem to the user, instead of closing the program.